goal_service/application/containers.py

```diff
@@ -11,18 +11,29 @@
     DevelopmentProgressionInstrumentation
 from goal_service.application.instrumentation.progression.metrics import \
     InMemoryProgressionMetrics
-from goal_service.infrastructure.orm import database
+from goal_service.infrastructure.config import Config
+from goal_service.infrastructure.orm.database import SqlAlchemy
+from goal_service.infrastructure.orm.mapping import DatabaseMapping
+from goal_service.infrastructure.orm.schema import DatabaseSchema
 
 
 class Core(containers.DeclarativeContainer):
-    session = database.session()
+    config = providers.Configuration('config')
+    config.override(Config(['common.yaml', 'environment.yaml']).read())
 
 
-class Queries(containers.DeclarativeContainer):
-    list_open_goals = providers.Factory(
-        ListOpenGoalsQuery, session=Core.session)
-    list_progressions = providers.Factory(
-        ListProgressionsQuery, session=Core.session)
+class Gateway(containers.DeclarativeContainer):
+    database = SqlAlchemy(
+        driver=Core.config.orm.driver(),
+        user=Core.config.orm.user(),
+        password=Core.config.orm.password(),
+        host=Core.config.orm.host(),
+        port=Core.config.orm.port(),
+        database=Core.config.orm.database())
+    database_schema = DatabaseSchema(database)
+    database_mapping = DatabaseMapping(
+        goal_table=database_schema.goal,
+        goal_progression_table=database_schema.goal_progression)
```

Comment:
Add complete schema to the mapping so when either of the two changes, we don't have to change the IoC container.